packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,72 @@
+import { isEmpty } from 'lodash'
+
+export default class {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources, target, force }) {
+    const targetHost = this._xo.getObject(target.master)
+    const _sources = sources.map(sourceId =>
+      this._xo.getObject(sourceId, 'pool')
+    )
+    let source
+    let sourceHost
+    for (source of _sources) {
+      sourceHost = this._xo.getObject(source.master)
+      if (sourceHost.productBrand !== targetHost.productBrand) {
+        throw new Error(
+          `a ${sourceHost.productBrand} pool cannot be merged into a ${
+            targetHost.productBrand
+          } pool`
+        )
+      }
+    }
+
+    const requiredPatchesSources = {}
+    let requiredPatchesTarget = []
+    for (source of _sources) {
+      requiredPatchesTarget = requiredPatchesTarget.concat(
+        this._xo.getPatchesDifference(source.master, target.master)
+      )
+      requiredPatchesSources[source.id] = this._xo.getPatchesDifference(
+        target.master,
+        source.master
+      )
```

Comment:
Let's say you want to merge `[A, B]` into `C`.
`B` has patch `1` installed, `A` and `C` don't have any.
When you'll loop over `[A, B]`, you'll first compare `A` and `C` (none of them have patches installed so you don't do anything) then you'll compare `B` and `C` and you'll notice that `C` needs to install patch `1`. But you'll never notice that `A` also needs to install it, now.